app_dart/lib/src/request_handlers/push_engine_status_to_github.dart

```diff
@@ -100,10 +100,21 @@ class PushEngineStatusToGithub extends ApiRequestHandler<Body> {
 
   String _getLatestStatus(List<LuciTask> tasks) {
     for (LuciTask task in tasks) {
-      if (task.status == Task.statusFailed) {
-        return GithubBuildStatusUpdate.statusFailure;
+      if (task.ref != 'refs/heads/master') {
+        log.debug('Skipping ${task.status} from commit ${task.commitSha} ref ${task.ref} builder ${task.builderName}');
+        continue;
+      }
+      switch (task.status) {
+        case Task.statusFailed:
+          log.debug('Using ${task.status} from commit ${task.commitSha} ref ${task.ref} builder ${task.builderName}');
+          return GithubBuildStatusUpdate.statusFailure;
+        case Task.statusSucceeded:
+          log.debug('Using ${task.status} from commit ${task.commitSha} ref ${task.ref} builder ${task.builderName}');
+          return GithubBuildStatusUpdate.statusSuccess;
```

Comment:
Don't we still want to loop through everything in `tasks` if the first one is green?

Reply:
The current logic is very inefficient:

1.- get the list of prod builders
2.- get the last 40 build for each builder
3.- run each builder through _getLatestStatus

The current logic will get the first pass or fail for each builder and set a failure status if one builder is red

Reply:
Ideally later on we may want to refactor this logic to calculate per commit instead of per builders.

Reply:
Ahhhh, I see, that makes sense. Can you add a dartdoc to this method, explicitly calling out that it takes in the tasks for a particular builder. LGTM!